website/addons/s3/static/s3UserConfig.js

```diff
@@ -0,0 +1,205 @@
+                error: error
+            });
+        });
+        return request;
+    };
+
+    /** Send POST request to authorize S3 */
+    self.sendAuth = function() {
+        // Selection should not be empty
+        if( !self.accessKey() && !self.secretKey() ){
+            self.changeMessage("Please enter both an API access key and secret key.", 'text-danger');
+            return;
+        }
+
+        if (!self.accessKey() ){
+            self.changeMessage("Please enter an API access key.", 'text-danger');
+            return;
+        }
+
+        if (!self.secretKey() ){
+            self.changeMessage("Please enter an API secret key.", 'text-danger');
+            return;
+        }
+        return osfHelpers.postJSON(
+            self.account_url,
+            ko.toJS({
+                access_key: self.accessKey,
+                secret_key: self.secretKey,
+            })
+        ).done(function() {
+            self.clearModal();
+            $modal.modal('hide');
+            self.updateAccounts();
+
+        }).fail(function(xhr, textStatus, error) {
+            var errorMessage = (xhr.status === 401) ? language.authInvalid : language.authError;
+            self.changeMessage(errorMessage, 'text-danger');
+            Raven.captureMessage('Could not authenticate with S3', {
+                url: self.account_url,
+                textStatus: textStatus,
+                error: error
+            });
+        });
+    };
+
+    self.askDisconnect = function(account) {
+        var self = this;
+        bootbox.confirm({
+            title: 'Disconnect Amazon S3 Account?',
+            message: '<p class="overflow">' +
+                'Are you sure you want to disconnect the S3 account <strong>' +
+                account.name + '</strong>? This will revoke access to S3 for all projects associated with this account.' +
+                '</p>',
+            callback: function (confirm) {
+                if (confirm) {
+                    self.disconnectAccount(account);
+                }
+            },
+            buttons:{
+                confirm:{
+                    label:'Disconnect',
+                    className:'btn-danger'
+                }
+            }
+        });
+    };
+
+    self.disconnectAccount = function(account) {
+        var self = this;
+        var url = '/api/v1/oauth/accounts/' + account.id + '/';
+        var request = $.ajax({
+            url: url,
+            type: 'DELETE'
+        });
+        request.done(function(data) {
+            self.updateAccounts();
+        });
+        request.fail(function(xhr, status, error) {
+            Raven.captureMessage('Error while removing addon authorization for ' + account.id, {
+                url: url,
+                status: status,
+                error: error
+            });
+        });
+        return request;
+    };
+
+    /** Change the flashed status message */
+    self.changeMessage = function(text, css, timeout) {
```

Comment:
Can you include the `ChangeMessageMixin` in this view model instead of re-defining the change message functions? https://github.com/CenterForOpenScience/osf.io/blob/develop/website/static/js/changeMessage.js#L11